packages/client/src/pages/Agent/AgentShow.re

```diff
@@ -1,56 +1,80 @@
+module ReducerConfig = {
+  type itemType = Context.Primary.Measurement.t;
+  type callFnParams = string;
+
+  let getId = (e: Context.Primary.Measurement.t) => e.id;
+  let callFn = (e: callFnParams) =>
+    Foretold__GraphQL.Queries.AgentMeasurements.componentWithMeasurementConnection(
+      ~id=e,
+    );
+
+  let isEqual = (a: itemType, b: itemType) => {
+    a.id == b.id;
   };
+};
 
-let component = ReasonReact.statelessComponent("AgentShow");
+module Reducer = SelectWithPaginationReducerFunctor.Make(ReducerConfig);
 
+let component = ReasonReact.statelessComponent("AgentShow");
 type pageParams = {id: string};
-let make = (~pageParams, ~layout=SLayout.FullPage.makeWithEl, _children) => {
+
+let make =
+    (
+      ~pageParams: pageParams,
+      ~loggedInUser: Context.Primary.User.t,
+      ~layout=SLayout.FullPage.makeWithEl,
+      _children,
+    ) => {
   ...component,
-  render: _ =>
-    Queries.Agent.component(~id=pageParams.id, ({agent, measurables}) =>
+  render: _ => {
+    Reducer.make(
+      ~itemsPerPage=20,
+      ~callFnParams=pageParams.id,
+      ~subComponent=selectWithPaginationParams =>
       SLayout.LayoutConfig.make(
-        ~head="" |> ste,
+        ~head=
+          switch (selectWithPaginationParams.selection) {
+          | Some(_selection) =>
+            <>
+              {Reducer.Components.deselectButton(
+                 selectWithPaginationParams.send,
+               )}
+              {Reducer.Components.correctButtonDuo(selectWithPaginationParams)}
+            </>
+          | None => <div />
+          },
         ~body=
-          measurables
-          |> E.L.fmap((m: Context.Primary.Measurable.t) => {
-               let measurements = m.measurements |> E.O.default([]);
-               <>
-                 <div className=block>
-                   {C.Measurable.Items.link(~m)}
-                   <C.Measurable.StatusDisplay measurable=m />
-                 </div>
-                 {measurements |> C.Measurements.Table.make}
-               </>;
-             })
-          |> E.A.of_list
-          |> ReasonReact.array,
+          switch (
+            selectWithPaginationParams.response,
+            selectWithPaginationParams.selection,
+          ) {
+          | (_, Some(measurement)) =>
+            switch (measurement.measurable) {
+            | Some(measurable) =>
+              <C.Measurable.FullPresentation id={measurable.id} loggedInUser />
+            | _ => <div />
+            }
+
+          | (Success(connection), None) =>
+            let onSelectClb = (e: Client.Context.Primary.Measurement.t) => {
+              Reducer.Components.sendSelectItem(
+                selectWithPaginationParams,
+                e.id,
+              );
+            };
+
+            let measurementsList = connection.edges |> Array.to_list;
+
+            C.Measurements.Table.makeAgentPredictionsTable(
+              ~measurementsList,
+              ~onSelect=onSelectClb,
+              (),
```

Comment:
I've noticed when scrolling through these that the order is wrong. When I click on one measurement, and then click the "next" icon on the top right, it doesn't bring me to the measurement/measurable that was next in the table. If it's an easy fix, I recommend making it.

Reply:
Sure, thanks.

Reply:
Fixed.